presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardRecoveryManager.java

```diff
@@ -145,15 +148,19 @@ private void enqueueMissingShards()
     void restoreFromBackup(UUID shardUuid)
     {
         File storageFile = storageService.getStorageFile(shardUuid);
-        if (storageFile.exists()) {
-            return;
-        }
-
         File backupFile = storageService.getBackupFile(shardUuid);
         if (!backupFile.exists()) {
             throw new PrestoException(RAPTOR_RECOVERY_ERROR, "No backup file found for shard: " + shardUuid);
         }
 
+        if (storageFile.exists()) {
+            if (storageFile.length() == backupFile.length()) {
+                return;
+            }
+            log.warn("Local shard file is corrupt, deleting it: %s", shardUuid);
```

Comment:
Use separate sentences here like the log message at the end of the function: "Local shard file is corrupt. Deleting local file: %s"